tests/Faithlife.Json.Tests/SnakeCasePropertyNamesContractResolverTests.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+using Faithlife.Json.ContractResolvers;
+using NUnit.Framework;
+using Newtonsoft.Json;
+
+namespace Faithlife.Json.Tests
+{
+	[TestFixture]
+	public class SnakeCasePropertyNamesContractResolverTests
+	{
+		[Test]
+		public void UsingStringComparisons()
+		{
+			var result = JsonConvert.DeserializeObject<Model>(
+				@"{
+					""regular_case"": 7,
+					""with_acronym_at_end"": 11,
+					""withacronym_at_beginning"": 13,
```

Comment:
Are we sure about this? I get that we want `OAuth` to be `oauth`, but does this work (or not work) for other examples?